liquidprompt

```diff
@@ -1550,6 +1556,70 @@ _lp_http_proxy_color() {
     lp_http_proxy_color="${LP_COLOR_PROXY}${LP_MARK_PROXY}${NO_COL}"
 }
 
+_lp_env_vars() { # [color_set, [color_unset]]
+    # Expects LP_ENV_VARS to be an array containing items of the form:
+    # "<ENV_VAR_NAME> <string if set>[ <string if unset>]"
+    # Strings may be `%s`, which will be replaced by the variable's actual content.
+
+    (( LP_ENABLE_ENV_VARS )) || return 2
+    [[ -n "${LP_ENV_VARS-}" && ${#LP_ENV_VARS} -gt 0 ]] || return 1
+
+    local color_set="${1-}"
+    local color_unset="${2-}"
+    local color_no="${NO_COL}"
+    if [[ -z "$color_set" && -z "$color_unset" ]]; then
+        color_no=""
+    fi
+
+    local var_rep var evar fmt def
+
+    lp_env_vars=()
+    # For all user-defined setup.
+    for var_rep in ${LP_ENV_VARS[@]+"${LP_ENV_VARS[@]}"}; do
+        IFS=' ' read -r var fmt def <<<"$var_rep"
+        # Variable name and format has to be set, but default is optional.
+        if [[ -n "${var}" && -n "${fmt}" ]]; then
+            # A way to test for undefined ${!var}
+            if declare -xp "${var}" &>/dev/null; then
```

Comment:
A better way to test this (that we normally use) that doesn't require a redirect is `[[ -n ${var+x} ]]`, which returns true if the variable is set. It is also much faster:

```bash
$ test() { for i in {1..100000}; do [[ -n ${var+x} ]]; done }
$ time test
real   	0m0.257s
user   	0m0.247s
sys    	0m0.008s

$ test() { for i in {1..100000}; do declare -xp "${var}" &>/dev/null; done }
$ time test
real   	0m1.746s
user   	0m1.083s
sys    	0m0.635s
```

Reply:
But this does not test if the indirected variable is set, it tests if the primary variable is set:
```sh
$ var="UNSET" ; if [[ -n ${var+x} ]]; then echo "SET"; else echo "UNSET"; fi
SET
$ var="UNSET" ; if declare -xp "${var}" &>/dev/null; then echo "SET"; else echo "UNSET"; fi
UNSET
```
And given that the *user* gives the names to watch, we need the second option. Bash >4.3 has nameref that would help, but before that, the only legit option seems to be indirection expansion, which does not work here.

Reply:
Ah, true. But the indirection expansion does work for me:
```bash
 $ foo=bar
$ var=foo
$ printf '%s\n' "${!var+x}"
x
$ unset foo
$ printf '%s\n' "${!var+x}"


```
So `[[ -n ${!var+x} ]]` should work.

Reply:
Indeed. I would have bet I tried that…